pybop/models/base_model.py

```diff
@@ -678,18 +698,38 @@ def approximate_capacity(self, inputs: Inputs):
     def built_model(self):
         return self._built_model
 
+    @built_model.setter
```

Comment:
Let's remove these additions setters of private properties until we have a safe way for a user to update them. I have removed the `parameter_set.setter` in #425.

Reply:
I've removed the ones that we currently don't require. The ones left are required within `FittingProblem` to build the model.  Let's refactor to clear these up where possible in a seperate PR.